File: API/_archive/GraphQL/python/01_inbuilt_basic.py
# BASIC EXAMPLE: Python Inbuilt http.server
# Demonstrates how GraphQL fundamentally runs over standard HTTP POST.
import json
from http.server import BaseHTTPRequestHandler, HTTPServer

class RequestHandler(BaseHTTPRequestHandler):
    def do_POST(self):
        if self.path == "/graphql":
            length = int(self.headers.get('content-length', 0))
            body = json.loads(self.rfile.read(length))
            
            # Simulated parsing of a basic GraphQL query string
            # e.g. {"query": "query { getUser { name } }"}
            query = body.get("query", "")
            if "getUser" in query:
                response = {"data": {"getUser": {"name": "Alice"}}}
            else:
                response = {"errors": [{"message": "Unknown query or mutation"}]}
                
            self.send_response(200)
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps(response).encode())

if __name__ == "__main__":
    print("Simulated GraphQL running on 8080...")
    HTTPServer(('', 8080), RequestHandler).serve_forever()

File: API/_archive/GraphQL/golang/01_inbuilt_basic.go
// BASIC EXAMPLE: Go Inbuilt net/http
// Demonstrates how a GraphQL server receives queries via HTTP POST.
package main

import (
	"encoding/json"
	"net/http"
	"strings"
)

func main() {
	http.HandleFunc("POST /graphql", func(w http.ResponseWriter, r *http.Request) {
		var req struct {
			Query string `json:"query"`
		}
		json.NewDecoder(r.Body).Decode(&req)

		w.Header().Set("Content-Type", "application/json")
		
		// Simulated Parsing
		if strings.Contains(req.Query, "getUser") {
			w.Write([]byte(`{"data": {"getUser": {"name": "Alice"}}}`))
		} else {
			w.Write([]byte(`{"errors": [{"message": "Unknown query"}]}`))
		}
	})
	http.ListenAndServe(":8080", nil)
}
